### ISSUE TITLE
Parallel Python interface significantly slower.

### ISSUE THREAD

**Issue by mariusCZ** [NanoComp/meep#1420]
Hello,

I am running meep version of 1.14 and Python 3.5.1, with the Python dependencies such as h5py, mpi4py installed using pip.

I am currently seeing significantly slower simulation times between, specifically in time steps per second. Everything in the simulation set up is completely identical, yet the time step for Python is average 0.024 and for Scheme it is 0.010, which is like losing a plane of symmetry. I also saw the same results with installing meep through conda packages.

The simulations are run on 5 nodes with 100 cores. The cell size is 32 x 25.3 x 6 and the resolution is 10. Open MPI version 1.10.2 is used.

Is it expected for Python interface to be slower? If not, where could I have gone wrong to see such dramatic decrease in simulation speed? I attached both the ctl and py files for reference. Thank you for the help.
[sims.tar.gz](https://github.com/NanoComp/meep/files/5503883/sims.tar.gz)

**Comment by mariswow** [comment]
Tried installing latest version of meep and also compiling mpi4py from source, but that did not change anything. Could it be related to the older Open MPI version?

**Comment by mariswow** [comment]
Just to give more information, the HPC I am using has computing nodes, which have 2x 14-core Intel Broadwell E5-2680v4 processors. I will try upgrading to an Open MPI version of 3.x.

In case this still does not work, is there a possibility to output the location of boundaries, sources and etc. within an HDF5 file using the Scheme interface? If it is not, I have a bit of experience with C++, how tough would it be to implement such function?

**Comment by stevengj** [comment]
Since the core calculation is all done in the same C++ library regardless of whether you use the Scheme or Python interface (and your simulation script does not use any expensive callbacks to Python on every iteration), I'm not sure what could be going on (unless you have accidentally used different simulation parameters in the two cases, but they look the same at first glance).   Are you using the same MPI version for both Scheme and Python?

**Comment by mariswow** [comment]
Yes, I am using the same MPI version for both of them. The cluster works with modules, meaning that only one version of MPI can be present at a single time.

I have tried disabling all graphical outputs, thinking that perhaps some I/O is causing bottleneck, but it did not help. However, there is one interesting thing that I noticed. Scheme has no errors when running, but when I run the simulation with Python, I get `Warning: grid volume is not an integer number of pixels; cell size will be rounded to nearest pixel.
 I also get ``/trinity/clustervision/CentOS/7/apps/python/3.5.1/lib/python3.5/site-packages/h5py/__init__.py:34: FutureWarning: Conversion of the second argument of issubdtype from float to np.floating is deprecated. In future, it will be treated as np.float64 == np.dtype(float).type. from ._conv import register_converters as _register_converters ``

However, the last warning is due to improper environment config that I have right now. The speed is still slower without it.
